### PR TITLE
Added Sequence Operations, selector and test cases

### PR DESCRIPTION
**Pull request by Vishalk91-4**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y 

re #2 

#### 2. What is the scope of this PR (e.g. component or file name):

- `grammar.js`
- `expr.txt`
- `assign.txt`
- `schema.txt`
- `grammar.json`
- `parser.c`

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [ ] Affects user behaviors
- [x] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

- [x]  Adding sequence operations
- [x]  Adding selectors
- [x] Adding select_suffix syntax
- [x]  Adding Test Cases for selector expression and sequence operations

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [ ] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

### CODE REVIEW

File: grammar.js [hunk 1]
```diff
@@ -717,6 +718,21 @@ module.exports = grammar({
...
+      $.concatenation,
+      $.subscript,
+      $.min,
+      $.max
+    )),
+    
+    in_operation: $ => prec.left(3, seq(choice($.list_comprehension, $.dictionary_comprehension, $.list, $.dictionary), 'in', choice($.list_comprehension, $.dictionary_comprehension, $.list, $.dictionary))),
+    not_in_operation: $ => prec.left(11, seq(choice($.list_comprehension, $.dictionary_comprehension, $.list, $.dictionary), 'not', 'in', $.expression)),
+    concatenation: $ => prec.left(12, seq(choice($.list_comprehension, $.dictionary_comprehension, $.list, $.dictionary), '+', $.expression)),
+    min: $ => prec.left(13, seq('min', '(', choice($.list_comprehension, $.dictionary_comprehension, $.list, $.dictionary), ')')),
```
Comment (Peefy) [inline grammar.js]: min and max may be call_expr?

Comment (Vishalk91-4) [reply grammar.js]: Yes
That did cross my mind after I added
Will remove that in the nxt commits

File: grammar.js [hunk 2]
```diff
@@ -717,8 +725,20 @@ module.exports = grammar({
...
+      $.in_operation,
+      $.not_in_operation,
+      $.concatenation,
+      $.subscript,
+      $.call,
+    ))),
+    
+    in_operation: $ => prec.left(3, seq(choice($.list_comprehension, $.dictionary_comprehension, $.list, $.dictionary), 'in', choice($.list_comprehension, $.dictionary_comprehension, $.list, $.dictionary))),
+    not_in_operation: $ => prec.left(11, seq(choice($.list_comprehension, $.dictionary_comprehension, $.list, $.dictionary), 'not', 'in', $.expression)),
+    concatenation: $ => prec.left(12, seq(choice($.list_comprehension, $.dictionary_comprehension, $.list, $.dictionary), '+', $.expression)),
```
Comment (Peefy) [inline grammar.js]: Put `x + y` into the binary expression grammar and consider `x - y`, `x * y`, and `x ** y`, etc, as well as their priorities.

Comment (Vishalk91-4) [reply grammar.js]: In Binary expression I found that it already has `x+y`, `x-y`, `x*y`, `x**y`
So, Do you want me to add `binary_operator` inside `sequence_operation` inplace of `concatenation` to cover it all

Comment (Peefy) [reply grammar.js]: Ok, you can remove concatenation here.

**Review by Peefy — APPROVED** [review]
LGTM

**Pull request merged**